File: packages/server/src/routes/queues.ts
import { Hono } from "hono";
import { getCachedQueueDetails } from "../lib/cache.js";

const app = new Hono();

// GET /api/queues
app.get("/", async (c) => {
  const details = await getCachedQueueDetails();
  return c.json(details);
});

export default app;

File: services/celery-gateway/src/celery_gateway/routers/queues.py
from __future__ import annotations

from typing import Any

from fastapi import APIRouter, Request

from ..models.tasks import FrontendQueueDetailsResult

router = APIRouter(tags=["queues"])


@router.get("/queues", response_model=FrontendQueueDetailsResult)
async def get_queues(request: Request) -> Any:
    cache = request.app.state.celery_cache
    return await cache.get("queue-details")
